spring-cloud-context/src/main/java/org/springframework/cloud/bootstrap/config/PropertySourceBootstrapConfiguration.java

```diff
@@ -241,3 +254,32 @@ private void handleIncludedProfiles(ConfigurableEnvironment environment) {
 	}
 
 }
+
+class BootstrapPropertySource<T> extends EnumerablePropertySource<T> {
+
+	private PropertySource<T> p;
+
+	BootstrapPropertySource(PropertySource<T> p) {
+		super(BOOTSTRAP_PROPERTY_SOURCE_NAME + "-" + p.getName(), p.getSource());
+		this.p = p;
+	}
+
+	@Override
+	public Object getProperty(String name) {
+		return p.getProperty(name);
+	}
+
+	@Override
+	public String[] getPropertyNames() {
+		Set<String> names = new LinkedHashSet<>();
+		if (!(this.p instanceof EnumerablePropertySource)) {
+			throw new IllegalStateException(
```

Comment:
Should it fail or log a warning?

Reply:
I took this code from the implementation in `CompositePropertySource` it just throws the exception, so figured I would do the same